plugin/pkg/admission/limitranger/admission.go

```diff
@@ -612,3 +589,154 @@ func PodValidateLimitFunc(limitRange *corev1.LimitRange, pod *api.Pod) error {
 	}
 	return utilerrors.NewAggregate(errs)
 }
+
+type podResourcesOptions struct {
+	// InPlacePodVerticalScalingEnabled indicates that the in-place pod vertical scaling feature gate is enabled.
+	InPlacePodVerticalScalingEnabled bool
+	// SidecarContainersEnabled indicates that the sidecar containers feature gate is enabled
+	SidecarContainersEnabled bool
+}
+
+// podRequests is a simplified version of pkg/api/v1/resource/PodRequests that operates against the core version of
```

Comment:
I couldn't figure out a better way to implement this, other than just duplicating the calculations.